integration-tests/protocolv2_test.go

```diff
@@ -0,0 +1,171 @@
+package integration_tests
+
+import (
+	"context"
+	"fmt"
+	"github.com/datastax/go-cassandra-native-protocol/client"
+	"github.com/datastax/go-cassandra-native-protocol/datatype"
+	"github.com/datastax/go-cassandra-native-protocol/frame"
+	"github.com/datastax/go-cassandra-native-protocol/message"
+	"github.com/datastax/go-cassandra-native-protocol/primitive"
+	"github.com/datastax/zdm-proxy/integration-tests/setup"
+	"github.com/stretchr/testify/require"
+	"net"
+	"testing"
+)
+
+func TestProtocolV2Connect(t *testing.T) {
+	originAddress := "127.0.0.2"
+	targetAddress := "127.0.0.3"
+
+	serverConf := setup.NewTestConfig(originAddress, targetAddress)
+	proxyConf := setup.NewTestConfig(originAddress, targetAddress)
+	proxyConf.ControlConnMaxProtocolVersion = "3" // simulate protocol downgrade to V2
```

Comment:
why this? if the server only supports v2 already then let's keep this as the default (which will be used by 99% of users) unless we want to actually test if this config setting works